crates/proto/src/tcp/tcp_client_stream.rs

```diff
@@ -41,13 +44,13 @@ impl<S: Connect + 'static + Send> TcpClientStream<S> {
     ///
     /// * `name_server` - the IP and Port of the DNS server to connect to
     #[allow(clippy::new_ret_no_self)]
-    pub fn new(
+    pub fn new<TE: 'static + Time>(
```

Comment:
It's unfortunate that we have to thread this type all the way through... But I don't know a better way at the moment.

Reply:
I'd think I'd like to expose this to the other crate. Ex. I plan to make this type abstract trait  in resolver crate.